src/ray/rpc/grpc_server.cc

```diff
@@ -140,23 +143,30 @@ void GrpcServer::PollEventsFromCompletionQueue(int index) {
         server_call->OnReplySent();
         // The rpc call has finished and can be deleted now.
         delete_call = true;
+        // A new call should be suplied.
+        need_new_call = true;
         break;
       default:
         RAY_LOG(FATAL) << "Shouldn't reach here.";
         break;
       }
     } else {
       // `ok == false` will occur in two situations:
-      // First, the server has been shut down, the server call's status is PENDING
-      // Second, server has sent reply to client and failed, the server call's status is
-      // SENDING_REPLY
+
+      // First, server has sent reply to client and failed, the server call's status is
+      // SENDING_REPLY.
       if (server_call->GetState() == ServerCallState::SENDING_REPLY) {
         server_call->OnReplyFailed();
+        // A new call should be suplied.
+        need_new_call = true;
       }
+
+      // Second, the server has been shut down, the server call's status is PENDING.
+      // And don't need to do anything other than deleting this call.
       delete_call = true;
```

Comment:
Why don't we just remove `delete_call `? I think setting `need_new_call = true` for shutdown is fine? (and it can simplify logic)

Reply:
No we can't, if we supply new calls when shutting down the gRPC server, it will block and can't be shut down. Because gRPC server only can be closed until no request in its CompleteionQueue.

Reply:
Sgtm